src/DynamoCore/Extensions/IExtensionLoader.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo.PackageManager;
+
+namespace Dynamo.Extensions
+{
+    /// <summary>
+    /// Handles loading extensions given an extension definition file
+    /// </summary>
+    internal interface IExtensionLoader
```

Comment:
The `ExtensionManager` will own one of these and will be handled one at startup.  The reason to do so is that `IExtensionLoader` can easily be mocked. Where possible, I try to mock any potential interaction with the filesystem (which are historically quite fragile).  This way, the implementor knows the intent.